hydrator-common/src/main/java/io/cdap/plugin/common/ReferenceBatchSink.java

```diff
@@ -41,4 +47,27 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     IdUtils.validateReferenceName(config.referenceName, collector);
     collector.getOrThrowException();
   }
+
+  /**
+   * Record field-level lineage for batch sink plugins (WriteOperation). This method should be called from prepareRun of
+   * any batch sink plugin.
+   *
+   * @param context BatchSinkContext from prepareRun
+   * @param outputName name of output dataset
+   * @param tableSchema schema of fields. Also used to determine list of field names. Schema and schema.getFields() must
+   * not be null.
+   * @param operationName name of the operation
+   * @param description operation description; complete sentences preferred
+   */
+  protected void recordLineage(BatchSinkContext context, String outputName, Schema tableSchema, String operationName,
+                               String description) {
+    Preconditions.checkNotNull(tableSchema, "schema for output %s is null.", outputName);
```

Comment:
Don't use Precondition, if will fail the pipeline. Should just do a if check and log as a warn message that field lineage will be missing